### PR TITLE
Adding support for execution cluster label

### PR DESCRIPTION
**Pull request by pmahindrakar-oss**
Signed-off-by: Prafulla Mahindrakar <prafulla.mahindrakar@gmail.com>

# TL;DR
This PR specifically addresses supporting execution cluster label through flytectl. 

- **GET**

Retrieves execution  cluster label for given project,domain combination or additionally with workflow name.

Here the command get execution cluster label for  project flytectldemo and development domain.

` flytectl get execution-cluster-label -p flytectldemo -d development `

eg : O/P
```
{"project":"flytectldemo","domain":"development","value":"buzz"}

```

Writing the execution cluster label to a file. If there are no execution cluster label, command would return an error
Here the command gets execution cluster label and writes the config file to ecl.yaml
eg:  content of ecl.yaml

` flytectl get execution-cluster-label --attrFile ecl.yaml`

```
.. code-block:: yaml

	domain: development
	project: flytectldemo
	value: foo
```

-  **UPDATE**
Updates execution cluster label for given project and domain combination or additionally with workflow name.

Updating the execution cluster label is only available  from config file
Here the command updates takes the input for execution cluster label from the config file ecl.yaml

eg:  content of ecl.yaml
`flytectl update execution-cluster-label -attrFile ecl.yaml`

- **DELETE**
Deletes execution cluster label for given project and domain combination or additionally with workflow name.
Deletes execution cluster label for project and domain
Here the command delete execution cluster label for  project flytectldemo and development domain.

` flytectl delete execution-cluster-label -p flytectldemo -d development `

Deleting execution cluster label using config file
Here the command deletes execution cluster label from the config file ecl.yaml
eg:  content of era.yaml which will use the project domain and workflow name for deleting the resource

 `flytectl delete execution-cluster-label --attrFile ecl.yaml`

```
.. code-block:: yaml

	domain: development
	project: flytectldemo
	value: foo
```

Deleting execution cluster label for a workflow
Here the command deletes execution cluster label for a workflow

```
flytectl delete execution-cluster-label -p flytectldemo -d development core.control_flow.run_merge_sort.merge_sort
```

## Type
 - [ ] Bug Fix
 - [X] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [X] Unit tests added
 - [X] Code documentation added
 - [ ] Any pending items have an associated Issue

## Tracking Issue
https://github.com/flyteorg/flyte/issues/992

## Follow-up issue
_NA_

### CODE REVIEW

File: cmd/delete/matchable_execution_cluster_label.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	sconfig "github.com/flyteorg/flytectl/cmd/config/subcommand"
+	"github.com/flyteorg/flytectl/cmd/config/subcommand/executionclusterlabel"
+	cmdCore "github.com/flyteorg/flytectl/cmd/core"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/admin"
+)
+
+const (
+	executionClusterLabelShort = "Deletes matchable resources of execution cluster label"
+	executionClusterLabelLong  = `
+Deletes execution clyuster label for given project and domain combination or additionally with workflow name.
```
Comment (katrogan) [inline cmd/delete/matchable_execution_cluster_label.go:16]:
```suggestion
Deletes execution cluster label for given project and domain combination or additionally with workflow name.
```

File: cmd/delete/matchable_execution_cluster_label.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+Deletes execution cluster label for project and domain
+Here the command delete execution cluster label for  project flytectldemo and development domain.
```
Comment (katrogan) [inline cmd/delete/matchable_execution_cluster_label.go:19]:
```suggestion
Here the command delete execution cluster label for project flytectldemo and development domain.
```

File: cmd/get/matchable_execution_cluster_label.go [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+ flytectl get execution-cluster-label -p flytectldemo -d development 
+
+eg : output from the command
+
+.. code-block:: json
+
+ {"project":"flytectldemo","domain":"development","value":"foo"}
+
+Retrieves execution cluster label for project and domain and workflow
+Here the command get execution cluster label for  project flytectldemo ,development domain and workflow core.control_flow.run_merge_sort.merge_sort
```
Comment (katrogan) [inline cmd/get/matchable_execution_cluster_label.go:31]:
```suggestion
Here the command get execution cluster label for project flytectldemo, development domain and workflow core.control_flow.run_merge_sort.merge_sort
```

File: cmd/update/matchable_execution_cluster_label.go [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    project: flytectldemo
+    value: foo
+
+::
+
+ flytectl update execution-cluster-label --attrFile ecl.yaml
+
+Updating execution cluster label for project and domain and workflow combination. This will take precedence over any other
+execution cluster label defined at project domain level.
+Update the execution cluster label for workflow core.control_flow.run_merge_sort.merge_sort in flytectldemo , development domain
```
Comment (katrogan) [inline cmd/update/matchable_execution_cluster_label.go:36]:
```suggestion
Update the execution cluster label for workflow core.control_flow.run_merge_sort.merge_sort in flytectldemo, development domain
```

File: docs/source/gen/flytectl_delete_execution-cluster-label.rst [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+---------------------------------------
+
+Deletes matchable resources of execution cluster label
+
+Synopsis
+~~~~~~~~
+
+
+
+Deletes execution clyuster label for given project and domain combination or additionally with workflow name.
```
Comment (katrogan) [inline docs/source/gen/flytectl_delete_execution-cluster-label.rst:13]:
```suggestion
Deletes execution cluster label for given project and domain combination or additionally with workflow name.
```

File: cmd/update/matchable_execution_cluster_label.go [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+)
+
+const (
+	executionClusterLabelShort = "Updates matchable resources of execution cluster label"
+	executionClusterLabelLong  = `
+Updates execution cluster label for given project and domain combination or additionally with workflow name.
+
+Updating to the execution cluster label is only available from a generated file. See the get section for generating this file.
+Also this will completely overwrite any existing custom label on project and domain and workflow combination.
+Would be preferable to do get and generate an attribute file if there is an existing attribute already set and then update it to have new values
```
Comment (katrogan) [inline cmd/update/matchable_execution_cluster_label.go:19]: is this advice necessary seeing as there can only be one label at a time?

**Review by katrogan — DISMISSED** [review]
Sorry this should have occurred to me earlier, but why the distinction between `-p project -d domain` and then positional args for the optional workflow name. Technically we can update matchable attributes for project, domain and launch plans too and if we expand flytectl support to that it would make the positional arg confusing

File: cmd/update/matchable_execution_cluster_label.go (hunk 6 above)
Comment (pmahindrakar-oss) [reply cmd/update/matchable_execution_cluster_label.go:19]: Yeah not required. removed it.

File: docs/source/gen/flytectl_delete_execution-cluster-label.rst (hunk 5 above)
Comment (pmahindrakar-oss) [reply docs/source/gen/flytectl_delete_execution-cluster-label.rst:13]: Done

File: cmd/update/matchable_execution_cluster_label.go (hunk 4 above)
Comment (pmahindrakar-oss) [reply cmd/update/matchable_execution_cluster_label.go:36]: Done

File: cmd/get/matchable_execution_cluster_label.go (hunk 3 above)
Comment (pmahindrakar-oss) [reply cmd/get/matchable_execution_cluster_label.go:31]: Done

File: cmd/delete/matchable_execution_cluster_label.go (hunk 2 above)
Comment (pmahindrakar-oss) [reply cmd/delete/matchable_execution_cluster_label.go:19]: Done

File: cmd/delete/matchable_execution_cluster_label.go (hunk 1 above)
Comment (pmahindrakar-oss) [reply cmd/delete/matchable_execution_cluster_label.go:16]: Done

**Comment by pmahindrakar-oss** [comment]
> Sorry this should have occurred to me earlier, but why the distinction between `-p project -d domain` and then positional args for the optional workflow name. Technically we can update matchable attributes for project, domain and launch plans too and if we expand flytectl support to that it would make the positional arg confusing

I didn't know about this requirement , But that change can be done by adding a flag to the command
- w workflowName
- l launplanName 
if both are passed it would be an error case.  And in that way we dont have to pass it as a positional argument.

But i think this can be deffered until we have this support for launchplans in admin . Let me know if that sounds ok.

**Review by katrogan — APPROVED** [review]
Sure deferring the launch plan support sounds totally okay!
